src/Form/FormLabel.d.ts

```diff
@@ -5,6 +5,7 @@ export interface FormLabelProps extends StandardProps<
   React.LabelHTMLAttributes<HTMLLabelElement>,
   FormLabelClassKey
 > {
+  component?: string | React.ComponentType<FormLabelProps>;
```

Comment:
I think this should actually be similar to [`Table`](https://github.com/smacpherson64/material-ui/blob/4f6832db00fc3db8c7bc813be281fbdafbf45e99/src/Table/Table.d.ts), i.e.

```ts
export interface FormLabelProps extends StandardProps<
  FormLabelBaseProps,
  FormLabelClassKey
> {
  component?: string | React.ComponentType<FormLabelBaseProps>;
  // ...
}

export type FormLabelBaseProps = React.LabelHTMLAttributes<HTMLLabelElement>;
```

because if you look at the implementation, only props of the base component type get passed on to the overriding component.